src/NodaTime.Web/Models/TimerCache.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading;
+
+namespace NodaTime.Web.Models
+{
+    public class TimerCache<T>
+    {
+        private readonly Timer timer;
+        private readonly Func<T> provider;
+        private readonly ILogger logger;
+        private readonly object padlock = new object();
```

Comment:
`lock` is a keyword, otherwise I'd use it. I suppose it could be `valueLock`. I tend to use `padlock` unless I'm working in a convention which has underscores...